clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -1060,6 +1063,12 @@ public int numStagedReceives(KafkaChannel channel) {
         return deque == null ? 0 : deque.size();
     }
 
+    public void updateClientInformation(KafkaChannel channel, ClientInformation clientInformation) {
```

Comment:
I am not 100% happy with this in the selector. Alternatively, we could make the metric accessible from the processor or have a listener mechanism to notify the selector when the client info is updated in the channel. What do you think?

The AVR being processed in many places is not ideal. It would be great if we could have it part of the establishment of the session in the future.